pageserver/src/consumption_metrics.rs

```diff
@@ -471,6 +461,72 @@ where
     current_metrics
 }
 
+#[derive(thiserror::Error, Debug)]
+enum UploadError {
+    #[error("server rejected metrics")]
+    Rejected(reqwest::StatusCode),
+    #[error("request failed")]
+    Reqwest(#[source] reqwest::Error),
+    #[error("cancelled")]
+    Cancelled,
+}
+
+impl UploadError {
+    fn is_reject(&self) -> bool {
+        matches!(self, UploadError::Rejected(_))
+    }
+}
+
+async fn upload(
+    client: &reqwest::Client,
+    metric_collection_endpoint: &reqwest::Url,
+    body: bytes::Bytes,
+    cancel: &CancellationToken,
+) -> Result<(), UploadError> {
+    if cancel.is_cancelled() {
```

Comment:
This still checks the cancellation token, your older branch didn't have it.

Reply:
Yeah... Oki, perhaps I'll try to cherry pick those.

Reply:
This sadly goes through the file splitting and git cannot follow those.

This is harmless flaw, I'll deal with it later.

Reply:
Backported this.